harness/determined/common/experimental/experiment.py

```diff
@@ -59,8 +60,18 @@ def _to_bindings(self) -> bindings.v1GetExperimentsRequestSortBy:
 class ExperimentOrderBy(enum.Enum):
     """Specifies whether a sorted list of experiments should be in ascending or
     descending order.
+
+    This class is deprecated in favor of ``OrderBy`` and will be removed in a future
+    release.
     """
 
+    warnings.warn(
```

Comment:
When does this actually get executed? I seem to be able to 

```python
experiment.ExperimentOrderBy.ASCENDING
```

without any warning (assume the same is true with `CheckpointOrderBy`).

Neither of my first couple ideas worked for printing a warning, so I'll just publish this comment for expedience and we'll come back as needed.